validator/rules/utils.go

```diff
@@ -0,0 +1,38 @@
+package rules
```

Comment:
Should be part of the core instead of rules

Reply:
utils is not good name for this

Reply:
I think we don't need this file, we can move it + traverse.go to validator.go

Reply:
exactly!